service/src/main/resources/db/migration/V262__child_absences_function.sql

```diff
@@ -0,0 +1,136 @@
+CREATE FUNCTION absences_in_period(childIds uuid[], period daterange)
+    RETURNS TABLE(
+                     id uuid,
+                     child_id uuid,
+                     date date,
+                     absence_type absence_type,
+                     modified_at timestamp with time zone,
+                     modified_by uuid,
+                     category absence_category,
+                     questionnaire_id uuid
+                 )
+AS $$
+BEGIN
+    RETURN QUERY
+        SELECT *
+        FROM absence a
+        WHERE a.child_id = ANY($1) AND between_start_and_end($2, a.date)
+        UNION
+        SELECT
+            dst.id as id,
+            dst.child_id,
+            d::date as date,
+            'OTHER_ABSENCE'::absence_type as absence_type,
+            dst.updated as modified_at,
+            null::uuid as modified_by,
+            'BILLABLE'::absence_category as category,
+            null::uuid as questionnaire_id
+        FROM daily_service_time as dst,
+             generate_series(lower($2), upper($2), '24 hours') as d
```

Comment:
You should use `upper($2) - 1` instead of `upper($2)`, because `upper` returns the exclusive end date of the date range (eg. `upper(daterange(current_date, current_date, '[]'))` is `current_date + 1`).